# Microbial Genomics Journey Workshop Gaborone 2024
## Part 10: Mobile genetic elements, Workflows and Pipelines

### updated knowledge map
We started this workshop with "Map your Microbial Genomics knowledge out". Knowledge mapping helps communicate information and solve complex problems. Now it is time to update your knowledge map with what we learnt in this workshop. Here is my version. You can access a powerpoint version of this Microbial genomics knowledge map [here](Knowledge_Map_2.pptx).<br/>
![Knowledge Map](Knowledge_Map_final.jpg)

---

### phastest
[phastest](https://phastest.ca/)
PHASTEST (PHAge Search Tool with Enhanced Sequence Translation) is the successor to the PHAST and PHASTER prophage finding web servers. PHASTEST is available online at https://phastest.ca, as an API for programmatic queries, or as a Docker image for local installations. You can check publication [here](https://academic.oup.com/nar/article/51/W1/W443/7167344).

### PlasmidFinder
[PlasmidFinder](https://cge.food.dtu.dk/services/PlasmidFinder/). This tool identifies plasmids in complete or draft sequenced isolates of bacteria. It can be accessed through this website or installed locally using docker. Instructions can be found [here](https://bitbucket.org/genomicepidemiology/plasmidfinder/src/master/).

#### Sunbeam
[Sunbeam](https://github.com/sunbeam-labs/sunbeam) is a pipeline written in snakemake by Kyle Bittinger team that simplifies and automates many of the steps in metagenomic sequencing analysis. It can be deployed on most Linux workstations and clusters. To read more, check out their paper in [Microbiome](https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-019-0658-x).

## Further Readings
* [DEFINITIONS REGARDING ISOLATE RELATIONSHIPS](https://www.sciencedirect.com/science/article/pii/S1198743X1463192X?via%3Dihub)
* [ChatGPT for bioinformatics](https://medium.com/@91mattmoore/chatgpt-for-bioinformatics-404c6d0817a1)
